tests/test_base.py

```diff
@@ -518,6 +518,10 @@ def test_components(self):
             ['1.4.5', '1.4.10-alpha', '1.4.10'],
             ['1.3.6', '1.4.4', '1.5.0'],
         ),
+        '~=1.4.0': (
```

Comment:
This test would fail previously, as for example `1.5.0` would be valid for the Spec.